webroot/src/pages/SelectPrivileges/SelectPrivileges.ts

```diff
@@ -0,0 +1,380 @@
+//
+//  SelectPrivileges.ts
+//  CompactConnect
+//
+//  Created by InspiringApps on 10/15/2024.
+//
+
+import { Component, Watch, mixins } from 'vue-facing-decorator';
+import { reactive } from 'vue';
+import { displayDateFormat } from '@/app.config';
+import MixinForm from '@components/Forms/_mixins/form.mixin';
+import InputCheckbox from '@components/Forms/InputCheckbox/InputCheckbox.vue';
+import InputButton from '@components/Forms/InputButton/InputButton.vue';
+import InputSubmit from '@components/Forms/InputSubmit/InputSubmit.vue';
+import Modal from '@components/Modal/Modal.vue';
+import { Compact } from '@models/Compact/Compact.model';
+import { FormInput } from '@/models/FormInput/FormInput.model';
+import { License, LicenseStatus } from '@/models/License/License.model';
+import { Licensee } from '@models/Licensee/Licensee.model';
+import { LicenseeUser } from '@/models/LicenseeUser/LicenseeUser.model';
+import { PrivilegePurchaseOption } from '@models/PrivilegePurchaseOption/PrivilegePurchaseOption.model';
+import { State } from '@/models/State/State.model';
+import moment from 'moment';
+
+@Component({
+    name: 'SelectPrivileges',
+    components: {
+        InputCheckbox,
+        InputSubmit,
+        InputButton,
+        Modal
+    }
+})
+export default class SelectPrivileges extends mixins(MixinForm) {
+    //
+    // Data
+    //
+    jurisprudencePendingConfirmation = '';
+
+    //
+    // Lifecycle
+    //
+    async created() {
+        await this.$store.dispatch('user/getPrivilegePurchaseInformationRequest');
+
+        if (this.alreadyObtainedPrivilegeStates?.length) {
+            this.initFormInputs();
+        }
+    }
+
+    //
+    // Computed
+    //
+    get purchaseDataList(): Array<PrivilegePurchaseOption> {
+        return this.currentCompact?.privilegePurchaseOptions || [];
+    }
+
+    get currentCompact(): Compact | null {
+        return this.userStore?.currentCompact || null;
+    }
+
+    get currentCompactCommissionFee(): number | null {
+        return this.currentCompact?.compactCommissionFee || null;
+    }
+
+    get userStore(): any {
+        return this.$store.state.user;
+    }
+
+    get user(): LicenseeUser | null {
+        return this.userStore.model;
+    }
+
+    get licensee(): Licensee | null {
+        return this.user?.licensee || null;
+    }
+
+    get userFullName(): string {
+        let name = '';
+
+        if (this.user) {
+            name = this.user.getFullName();
+        }
+
+        return name;
+    }
+
+    get privilegeList(): Array<License> {
+        return this.licensee?.privileges || [];
+    }
+
+    get licenseList(): Array<License> {
+        return this.licensee?.licenses || [];
+    }
+
+    get alreadyObtainedPrivilegeStates(): Array<string | null> {
+        return this.licenseList.concat(this.privilegeList).map((license) => license?.issueState?.abbrev || null);
```

Comment:
why not just filter out nulls?

Reply:
The ts compiler is seeming to struggle to comprehend the type change associated with using .filter so I've done this using a forEach. That is likely why I didn't do this in the first place.